server/client.go

```diff
@@ -333,17 +341,29 @@ func (c *client) GetTLSConnectionState() *tls.ConnectionState {
 // FIXME(dlc) - This is getting bloated for normal subs, need
 // to optionally have an opts section for non-normal stuff.
 type subscription struct {
+	nm      int64 // Will atomicall be set to -1 on unsub or connection close
 	client  *client
 	im      *streamImport   // This is for import stream support.
 	shadow  []*subscription // This is to track shadowed accounts.
 	subject []byte
 	queue   []byte
 	sid     []byte
-	nm      int64
 	max     int64
 	qw      int32
 }
 
+// Indicate that this subscription is closed.
+// This is used in pruning of some cache.
```

Comment:
pruning of route cache items.

Reply:
and gateway..

Reply:
Same cache?

Reply:
Same structure, but each route/gateway connection has its own.